wiki/Game_modifier/Flashlight/en.md

```diff
@@ -0,0 +1,76 @@
+
+All of this adds up to an effect that looks like a virtual flashlight is being shone on the player's cursor:
+
+![FL gameplay](img/GM_FL_O.jpg "Gameplay of osu!standard with the Flashlight mod")
+
+It should be noted that when paired with the Hidden mod, the "flashlight's" visible area barely matters: with limited vision, the fading of the hit objects may occur while the cursor is not focused to the hit object's appearance point.
+
+The Flashlight mod is widely regarded by many as the hardest mod in osu!standard, and scores obtained with it typically require players to memorize the entire beatmap.
+
+### osu!taiko
+
+In [osu!taiko](/wiki/Game_Modes/osu!taiko), the position of visible area is fixed at the hitting area. And similarly to osu!standard, the visible area shrinks as the combo increases: shrinking at 100x and 200x combo while returning to its original size if the combo breaks.
+
+![FL gameplay o!t](img/GM_FL_T.jpg "Gameplay of osu!taiko with the Flashlight mod")
+
+When paired with the Hidden mod, the "flashlight's" visible area becomes a moot point because notes are technically "invisible" because the notes fade out completely upon reaching the visible area. This also requires complete memorization of the beatmap.
+
+### osu!catch
+
+In [osu!catch](/wiki/Game_Modes/osu!catch), the Flashlight mod behaviour is the same as in osu!standard, except for the fact that the visible area follows the catcher instead of the cursor. And due to the nature of osu!catch, the visible area is notably larger than in osu!standard or osu!taiko. Although this does not mean it is any less challenging.
+
+![FL gameplay o!c](img/GM_FL_C.jpg "Gameplay of osu!catch with the Flashlight mod")
+
+When paired with the Hidden mod, the fruits are momentarily visible if the catcher is *directly under* the fruits. That is, up until the player reaches 100x combo, at which point the fruits become completely invisible by the time the fruits reach the visible area. Much like osu!standard and osu!taiko, this also requires complete memorization of the beatmap.
+
+### osu!mania
+
+In [osu!mania](/wiki/Game_Modes/osu!mania), the visible area is restricted to a relatively thin horizontal bar at the center of the screen whilst everything else is obstructed from view. Thus this could be thought of the Hidden mod and [Fade In](/wiki/Game_modifier/Fade_in) mod combined. (That is, without the changing sizes of the visible area.)
+
+![FL gameplay o!m](img/GM_FL_M.jpg "Gameplay of osu!mania with the Flashlight mod")
+
+## Trivia
+
+- If a beatmap is passed with an S or SS grade with the Hidden mod enabled, the beatmap will award the silver variant of the grade instead.
+- The Flashlight mod was originally subjected to heavy controversy regarding the implementation in 2010 for being the easiest mod to hack on forced the mod to be unranked until a patch was implemented to cover the Flashlight implementation loophole.
```

Comment:
> for being the easiest mod to hack on forced the mod to be unranked until

this part of the sentence reads weird to me

Reply:
Have no idea why the word "hack" is even there